### ISSUE TITLE
sigmoid forcing model

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/ospree#323]
I ran the sigmoidal model on forcing -- but I ended up 67 divergent transitions (given the iterations though I am not sure that is so bad). I re-ran with an even longer warmup but still got proportionally the same divergent transitions. However, the results were stable and the model diagnostics all look pretty good, making me think the model is all stable.  

If you look at the code (models_stan_supp.R) you'll see I did it cheaply by swapping the names in the data of chill and force -- so when you read the model output you need to swap the names there too. 

Here are the estimates from the all spp 15000, warmup=12000 ... 

<img width="547" alt="m2l nisig force" src="https://user-images.githubusercontent.com/2153556/69149001-68e7ae80-0a8a-11ea-9774-7bffa75b8404.png">

And here's the issue -- b_chill is the only parameter I saw with any obvious issues:

<img width="1108" alt="b_chill_m2l nisig force_allspp" src="https://user-images.githubusercontent.com/2153556/69149041-7d2bab80-0a8a-11ea-9275-6b60ef7f9641.png">

I am checking the main species model now, then will work on writing some updated text for letter and supp (will email or post as separate file-chunks to avoid conflicts).

**Comment by lizzieinvancouver** [comment]
I got the model to run with just 1 divergent transition using the main model species -- there is still a bit of multimodality in the bchill but seems more okay. Here's the screen shot after running this code:
```

datalist.bb.adjforce <- datalist.bb
names(datalist.bb.adjforce)
names(datalist.bb.adjforce) <- c("y", "force", "chill",  "photo", "sp", "N", "n_sp")
m2l.nisig.force = stan('stan/archive/nointer_2level_interceptonly_sigmoid.stan', data = datalist.bb.adjforce,
               iter = 15000, warmup=14000, control=list(adapt_delta=0.999))
# 67 divergent transitions given 15000, warmup=12000; 17 divergent transitions given above -- both for allsppmodel (3245 obs, 203 species)
# save(m2l.nisig.force, file="stan/output/m2l.nisig.force_allsppmodel.Rda") # too big

summary(m2l.nisig.force)$summary[c("b_force", "b_photo","a_chill", "b_chill"),]
observed.here <- bb.stan$resp
nonlin.sum<-summary(m2l.nisig.force)$summary 

preds.nonlin.sum <- nonlin.sum[grep("yhat", rownames(nonlin.sum)),]
nonlin.mod.R2 <- 1- sum((observed.here-preds.nonlin.sum[,1])^2)/sum((observed.here-mean(observed.here))^2)
summary(lm(preds.nonlin.sum[,1]~observed.here)) 
```

<img width="692" alt="m2l nisig forcemainmodel" src="https://user-images.githubusercontent.com/2153556/69202722-5061ad80-0af7-11ea-9a09-016adfaf3da4.png">

**Comment by lizzieinvancouver** [comment]
@AileneKane Sorry I did not get further on this! Here's what I hoped to add to letter:
- This is certainly an important point as non-linearities in forcing definitely occur and including these has been shown to improve models (cite Wang 2017 paper I added to bib). These non-linearities however often occur at very low temperatures or above 40 C, depending on the plant species (https://www.springer.com/gp/book/9783540435167). Most of our studies are in [give 25-75% values of force and maybe min and max], suggesting we are in the mainly linear part of the curve. 
- Also, we are not aware of any studies in OSPREE that varied forcing (given photo and chill held constant) and did not see an advance. 
- Given all this, we did not investigate non-linearities.
- Would be good to review Caffarra and Gauzere papers, and see if they mention any non-linearities in forcing -- if not, we should point this out too.
- Given these reviews we tried to fit a sigmoidal forcing model -- but gave similar results to our non-linear chill model (it did not improve model fit and gave non-sensical results for a_force and b_force).
- Would be good to say, if we can nicely -- there are lots of non-linear models for chilling, we employed two here, but we are not aware of such common models for forcing and would be happy to apply one to our data if the reviewer can suggest one.
